### PR TITLE
Fixes for various issues with PINNs' tutorials

### PR DESCRIPTION
**Pull request by brunopjacob**
This PR addresses the following issues:

- https://github.com/pnnl/neuromancer/issues/181
- https://github.com/pnnl/neuromancer/issues/179
- https://github.com/pnnl/neuromancer/issues/178
- https://github.com/pnnl/neuromancer/issues/177
- https://github.com/pnnl/neuromancer/issues/172
- https://github.com/pnnl/neuromancer/issues/186

### CODE REVIEW

File: examples/PDEs/Part_2_PINN_BurgersEquation.py
Comment (RBirmiwal) [inline examples/PDEs/Part_2_PINN_BurgersEquation.py:1]: I believe the .to(device) here is unnecessary: 
```
ell_f = scaling * (f_pinn == torch.tensor(0.).to(device)) ^ 2
```

i encountered this issue before and solved it by using decorator - handle_device_placement -  for our constraints as in 
```
class LT(nn.Module):
    """
    Less than constraint for upper bounding the left hand side by the right hand side.
    Used for defining infix operator for the Variable class and calculating constraint
    violation losses for the forward pass of Constraint objects.

    constraint: g(x) <= b
    forward pass returns:
        value = g(x) - b
        penalty = relu(g(x) - b)
        loss = torch.mean(penalty)
    """
    def __init__(self, norm=1):
        super().__init__()
        self.norm = norm

    def __str__(self):
        return 'lt'

    @handle_device_placement
    def forward(self, left, right):
        """

        :param left: torch.Tensor
        :param right: torch.Tensor
        :return: zero dimensional torch.Tensor, torch.Tensor, torch.Tensor
        """

        value = left - right
        penalty = F.relu(value)
        if self.norm == 2:
            penalty = penalty ** 2
        loss = torch.mean(penalty)
        return loss, value, penalty
```


Please confirm that this works, and if so, remove the .to(device) for the 0 tensor. I just think it looks much cleaner without and using this decorator. Thanks

File: examples/PDEs/Part_2_PINN_BurgersEquation.ipynb
Comment (RBirmiwal) [inline examples/PDEs/Part_2_PINN_BurgersEquation.ipynb:1]: please see the equivalent comment for the .py file for Burgers

File: examples/PDEs/Part_3_PINN_BurgersEquation_inverse.py
Comment (RBirmiwal) [inline examples/PDEs/Part_3_PINN_BurgersEquation_inverse.py:1]: same comment as Burgers

File: examples/PDEs/Part_3_PINN_BurgersEquation_inverse.ipynb
Comment (RBirmiwal) [inline examples/PDEs/Part_3_PINN_BurgersEquation_inverse.ipynb:1]: same comment as Burgers

File: examples/PDEs/Part_4_PINN_LaplaceEquationSteadyState.ipynb
Comment (RBirmiwal) [inline examples/PDEs/Part_4_PINN_LaplaceEquationSteadyState.ipynb:1]: Thank you for adding all the .to(device) calls here. Unforuntatley, and this is my opinion, there are 24 .to(device) lines added that the user needs to do to run this on the GPU. To me, this is **way** too many. I would like to see this wrapped in the Lightning framework and have a single argument in the LitTrainer(device=...) solve this whole issue. For the other notebooks the number of .to(device) calls was small so wasn't an issue.  

We can setup meeting to migrate Part4-6 to Lightning, it should be straightforward. Thanks

File: examples/PDEs/Part_5_Pendulum_Stacked.ipynb
Comment (RBirmiwal) [inline examples/PDEs/Part_5_Pendulum_Stacked.ipynb:1]: Please see comment for Part 4

File: examples/PDEs/Part_6_PINN_NavierStokesCavitySteady_KAN.ipynb
Comment (RBirmiwal) [inline examples/PDEs/Part_6_PINN_NavierStokesCavitySteady_KAN.ipynb:1]: Please see comment for Part 4

**Review by drgona — APPROVED** [review]
thanks Bruno!
I agree with Rahul to reduce the number of .to(device) lines, but as a hotfix this is good to me

**Comment by brunopjacob** [comment]
@RBirmiwal please do one more check, and if it looks good, it's good to merge. Thanks for the feedback.

**Pull request merged**